### PR TITLE
Optimize addOutput

### PR DESCRIPTION
**Pull request by Glease**
GTNewHorizons/GT-New-Horizons-Modpack#7339

Works with GT++ super bus, normal output bus and ME output bus. Does not crash the game if ME output bus is used without AE2 installed somehow.

### CODE REVIEW

**Review by draknyte1 — APPROVED** [review]
My buses literally just have more slots, no reason they wouldn’t be compatible.

File: src/main/java/gregtech/common/tileentities/machines/GT_MetaTileEntity_Hatch_OutputBus_ME.java [hunk 1]
```diff
@@ -60,6 +61,16 @@ public void onFirstTick(IGregTechTileEntity aBaseMetaTileEntity) {
...
     }
 
+    @Override
+    public boolean storeAll(ItemStack aStack) {
+        if (!GregTech_API.mAE2)
+            return false;
+        int tTotal = aStack.stackSize;
+        int tStored = store(aStack);
+        aStack.stackSize -= tStored;
+        return tTotal < tStored;
```
Comment (repo-alt) [inline src/main/java/gregtech/common/tileentities/machines/GT_MetaTileEntity_Hatch_OutputBus_ME.java:71]: Isn't that the same as `tStored > 0` ?

Comment (Glease) [reply src/main/java/gregtech/common/tileentities/machines/GT_MetaTileEntity_Hatch_OutputBus_ME.java:71]: hell wait no this is not going to work well

File: src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_OutputBus.java [hunk 2]
```diff
@@ -110,6 +110,42 @@ public Object getClientGUI(int aID, InventoryPlayer aPlayerInventory, IGregTechT
...
+     * If you need atomicity you should use {@link gregtech.api.interfaces.tileentity.IHasInventory#addStackToSlot(int, ItemStack)}
+     * @param aStack Assume valid.
+     *               Will be mutated.
+     *               Take over the ownership. Caller should not retain a reference to this stack if the call returns true.
+     * @return true if stack is fully accepted. false is stack is partially accepted or nothing is accepted
+     */
+    public boolean storeAll(ItemStack aStack) {
+        for (int i = 0, mInventoryLength = mInventory.length; i < mInventoryLength; i++) {
+            ItemStack tSlot = mInventory[i];
+            if (GT_Utility.isStackInvalid(tSlot)) {
```
Comment (repo-alt) [inline src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_OutputBus.java:124]: Isn't that inverse?

Comment (Glease) [reply src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_OutputBus.java:124]: `If (current slot is empty) {replace slot with input} else {try add input to slot}`

File: src/main/java/gregtech/common/tileentities/machines/GT_MetaTileEntity_Hatch_OutputBus_ME.java (hunk 1 above)
Comment (repo-alt) [reply src/main/java/gregtech/common/tileentities/machines/GT_MetaTileEntity_Hatch_OutputBus_ME.java:71]: ah, I see

File: src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_OutputBus.java (hunk 2 above)
Comment (repo-alt) [reply src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_OutputBus.java:124]: isStackInvalid == "is empty"?

Comment (Glease) [reply src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_OutputBus.java:124]: Yes?

Comment (repo-alt) [reply src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_OutputBus.java:124]: oh, right, I'm blind

**Pull request merged**